### DISCUSSION TITLE
Select all checkboxes with no-children

### DISCUSSION THREAD

**Discussion by bmuthusubash** [twilio-labs/paste#1873] [Q&A] [answered]
Hi Team, 

We have a use case of representing multiple checkboxes on the table and having few selectAll checkboxes on column level. Due to the nature of the checkbox, they will not have labels as it has to be inferred from the row names and column names.(PFA)

**Issue**: When `isSelectAll` is set to the checkbox that contains only ScreenRecorder component, the background colour that represent selectAll is not centered due to the presence of empty label & margin next to the checkbox. I understand this very minor, but looks bit off on the table as the selectAll checkbox appears in various levels and multiples. Is there any way to center them?

Also would like to understand if we would be able to update the size of the checkbox? I tried them on the sandbox below, but does not seem to work.

Sandbox: https://codesandbox.io/s/empty-platform-6ru6x?file=/src/index.tsx

![image](https://user-images.githubusercontent.com/84200882/133997428-2736f1a5-92ad-46ab-b797-3d64185c52d8.png)

**Comment by serifluous** [comment, marked as answer]
Hey @bmuthusubash ! This is something we should've caught with @pguptaDS in the original design discussion #1520 : The "Select all" checkboxes should have a text label. Otherwise, it's not clear what they do—I assume this is why tooltips were added in in the mockups.

Here's a [quick mockup](https://www.figma.com/file/SpeLpBoqN2vZv8KWcSn0J1/Feedback?node-id=718%3A41450) of what that might look like:

![image](https://user-images.githubusercontent.com/3821840/134384924-5a18d0ea-fd1d-4d82-aec2-67c67fe62531.png)

With the text labels added in, you might not need to center the content, but we can help out if that's still your preferred route. The sizing of the checkboxes however can't be changed. If you have insight from customers that the sizing is hurting the usability of this page, though, please let us know!
